operators/src/source/gdal_source/loading_info.rs

```diff
@@ -0,0 +1,535 @@
+    time_step_iter: TimeStepIter,
+    params: GdalDatasetParameters,
+    time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    step: TimeStep,
+    max_t2: TimeInstance,
+}
+
+impl DynamicGdalLoadingInfoPartIterator {
+    fn new(
+        time_step_iter: TimeStepIter,
+        params: GdalDatasetParameters,
+        time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+        step: TimeStep,
+        max_t2: TimeInstance,
+    ) -> Result<Self> {
+        // TODO: maybe fail on deserialization
+        if time_placeholders.is_empty()
+            || time_placeholders.keys().any(String::is_empty)
+            || time_placeholders
+                .values()
+                .any(|value| value.format.is_empty())
+        {
+            return Err(Error::DynamicGdalSourceSpecHasEmptyTimePlaceholders);
+        }
+
+        Ok(Self {
+            time_step_iter,
+            params,
+            time_placeholders,
+            step,
+            max_t2,
+        })
+    }
+}
+
+impl Iterator for DynamicGdalLoadingInfoPartIterator {
+    type Item = Result<GdalLoadingInfoPart>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let t1 = self.time_step_iter.next()?;
+
+        let t2 = t1 + self.step;
+        let t2 = t2.unwrap_or(self.max_t2);
+
+        let time_interval = TimeInterval::new_unchecked(t1, t2);
+
+        let loading_info_part = self
+            .params
+            .replace_time_placeholders(&self.time_placeholders, time_interval)
+            .map(|loading_info_part_params| GdalLoadingInfoPart {
+                time: time_interval,
+                params: loading_info_part_params,
+            });
+
+        Some(loading_info_part)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct NetCdfCfGdalLoadingInfoPartIterator {
+    time_step_iter: TimeStepIter,
+    params: GdalDatasetParameters,
+    step: TimeStep,
+    dataset_time_start: TimeInstance,
+    max_t2: TimeInstance,
+    band_offset: usize,
+}
+
+impl NetCdfCfGdalLoadingInfoPartIterator {
+    fn new(
+        time_step_iter: TimeStepIter,
+        params: GdalDatasetParameters,
+        step: TimeStep,
+        dataset_time_start: TimeInstance,
+        max_t2: TimeInstance,
+        band_offset: usize,
+    ) -> Self {
+        Self {
+            time_step_iter,
+            params,
+            step,
+            dataset_time_start,
+            max_t2,
+            band_offset,
+        }
+    }
+
+    // fn band_offset(reference_time: TimeInstance, step: TimeStep, input_time: TimeInstance) -> usize {
+    //     reference_time + step * band_index as TimeInstance
+    // }
+}
+
+impl Iterator for NetCdfCfGdalLoadingInfoPartIterator {
+    type Item = Result<GdalLoadingInfoPart>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let t1 = self.time_step_iter.next()?;
+
+        // our first band is the reference time
+        // TODO: what if it intersects the reference time with t1 < ref?
```

Comment:
if it intersects it should also produce data. the reference time is no hard bound.

Reply:
What should we return? The data is only partially valid and we cannot specify a band here.

Reply:
i thought we generate time intervals here?

Reply:
daf432b 

New answers and new questions.